main.ts

```diff
@@ -80,6 +80,30 @@ export default class RtlPlugin extends Plugin {
 				return;
 			this.switchDocumentDirection(view.editor, view);
 		});
+
+		// Because the indicator button won't show properly the direction text when startup
+		// or when changing between file, we need to register "file-open" view and
+		// set the callback to update the direction text.
+		this.registerEvent(this.app.workspace.on("active-leaf-change", leaf => {
+			let view = leaf.view;
+			if (view instanceof MarkdownView && view.file) {

```

Comment:
There is a redundancy here with the `updateStatusBar` function below. Can you make the two reuse the same code (maybe by sending the `updateStatusBar` an optional `view` argument)?